core/src/main/java/com/dtolabs/rundeck/core/jobs/JobReference.java

```diff
@@ -50,4 +50,14 @@
      * @return server UUID
      */
     String getServerUUID();
+
+    /**
+     * @return original job name in case of renaming
+     */
+    default String getOriginalJobName(){ return null; }
+
+    /**
+     * @return original group name in case of renaming
+     */
+    default String getOriginalGroupName(){ return null; }
```

Comment:
I think you should rename these to "originalQuartzJobName" and "originalQuartzGroupName" because it can be confusing on what "job" it is referring to.